### PR TITLE
Added update_content function to update a file

### PR DESCRIPTION
**Pull request by ahmad88me**
Implemented the below
https://developer.github.com/v3/repos/contents/#update-a-file

### CODE REVIEW

**Comment by jzelinskie** [comment]
Looks good, thanks for the PR.
Can you write a test for this function?

**Comment by ahmad88me** [comment]
Ok @jzelinskie , I will look into the tests and how they are structured and will add a test for it

**Comment by jzelinskie** [comment]
thanks

File: .settings/org.eclipse.core.resources.prefs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+eclipse.preferences.version=1
+encoding//github/ContentFile.py=utf-8
+encoding//github/File.py=utf-8
+encoding//github/GistFile.py=utf-8
```
Comment (henklaak) [inline .settings/org.eclipse.core.resources.prefs]: asdfasdfasdf

**Comment by cwittern** [comment]
I needed this, so I cloned @ahmad88me 's fork.  Then I realized that I also need a create_content function to implement "PUT /repos/:owner/:repo/contents/:path" from https://developer.github.com/v3/repos/contents/#create-a-file.
I would be happy to contribute this as well if you are interested, but how should this be done?  Should I first create a PR for @ahmad88me 's fork?
My commit is here: https://github.com/cwittern/PyGithub/commit/5948ca9f85d906c85dba508a62f84a4dcf9f14b2

**Comment by jzelinskie** [comment]
@cwittern yes, you could do that, but you'd have to rebase at some point for us to upstream the change.

**Comment by jayfk** [comment]
Is there any documentation to write tests for PyGithub? I wasn't able to create the right replay data to write a working test for this function.

**Comment by jayfk** [comment]
I've added tests, pep formatting, a proper return value and removed @ahmad88me's editor files in https://github.com/ahmad88me/PyGithub/pull/1

Once @ahmad88me has the time to merge the changes in his fork, the commits should be automatically attached here.

**Comment by jayfk** [comment]
I just merged https://github.com/PyGithub/PyGithub/pull/379 that implements the full content API.

**Pull request closed without merging**